packages/components/src/avatar/avatar.tsx

```diff
@@ -1,97 +1,356 @@
-import { useEffect, useState } from 'react'
+import { cloneElement, useMemo, useState } from 'react'
 
-import { Stack, styled, Text, Unspaced } from '@tamagui/core'
+import {
+  LockedIcon,
+  MembersIcon as MembersIcon12,
+  UnlockedIcon,
+} from '@status-im/icons/12'
+import { MembersIcon as MembersIcon16 } from '@status-im/icons/16'
+import { MembersIcon as MembersIcon20 } from '@status-im/icons/20'
+import { Stack, styled, Unspaced } from '@tamagui/core'
+import { Platform } from 'react-native'
 
 import { Image } from '../image'
+import { Text } from '../text'
+import { tokens } from '../tokens'
+import { generateIdenticonRing } from './utils'
 
-import type { GetStyledVariants } from '@tamagui/core'
+import type { TextProps } from '../text'
+import type { ColorTokens, GetStyledVariants } from '@tamagui/core'
 
 type Variants = GetStyledVariants<typeof Base>
 
-type Props = {
-  src: string
-  size: 80 | 56 | 48 | 32 | 28 | 24 | 20 | 16
-  shape?: Variants['shape']
-  outline?: Variants['outline']
-  indicator?: GetStyledVariants<typeof Indicator>['state']
-}
+type AvatarProps =
+  | {
+      type: 'user'
+      size: 80 | 56 | 48 | 32 | 28 | 24 | 20 | 16
+      name: string
+      src?: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+      indicator?: GetStyledVariants<typeof Indicator>['state']
+    }
+  | {
+      type: 'user'
+      size: 80 | 56 | 48 | 32
+      name: string
+      src?: string
+      outline?: Variants['outline']
+      backgroundColor?: ColorTokens
+      indicator?: GetStyledVariants<typeof Indicator>['state']
+      colorHash?: number[][]
```

Comment:
Can we please, do not distinguish between two user types? I don't think it is logically incorrect to allow it for every user Avatar.

Reply:
done at https://github.com/status-im/status-web/pull/371/commits/7cdc69111d1cb0e9891eafec09726801d5ed864f and https://github.com/status-im/status-web/pull/371/commits/cc723d7004a29e18a915f55f3d39429cf4a9b0ed